cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/AnomalyDetector.java

```diff
@@ -175,7 +179,7 @@ public void shutdown() {
     synchronized (_shutdownLock) {
       _shutdown = true;
     }
-    _anomalies.addFirst(SHUTDOWN_ANOMALY);
+    _anomalies.add(SHUTDOWN_ANOMALY);
```

Comment:
This will not necessarily become the first item in the queue in case there are other broker failures -- am I missing something?

Reply:
Here is an implicit assumption.
`SHUTDOWN_ANOMALY` is a `BrokerFailures` with detection time to be 0, based on the current comparator(first check priority then detection time), it will be added to the head to the queue.